### ISSUE TITLE
hail does not respect instance exclusion tags

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#855]
Originally reported of Google Code with ID 799.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

# gnt-cluster --version
gnt-cluster (ganeti v2.9.5) 2.9.5

# gnt-cluster version
Software version: 2.9.5
Internode protocol: 2090000
Configuration format: 2090000
OS api version: 20
Export interface: 0
VCS version: v2.9.5

<b>What distribution are you using?</b>
# cat /etc/*-release
SUSE Linux Enterprise Server 11 (x86_64)
VERSION = 11
PATCHLEVEL = 3

<b>What steps will reproduce the problem?</b>
1. set htools instance exclusion tags
# gnt-cluster info | grep htools
Tags: htools:iextags:customer, htools:iextags:sapsid

2. add instances with exclusion tag (placement via hail)
# gnt-instance add -t sharedfile ... --tags sapsid:xxx vm1
# gnt-instance add -t sharedfile ... --tags sapsid:xxx vm2

<b>What is the expected output? What do you see instead?</b>

Expected is, that hail places instances on nodes wrt exlusion tag. But it happend, that the same node was choosen:

2014-04-15 09:25:01,681: ganeti-masterd pid=22310/Jq16/Job890534/I_CREATE INFO Selected nodes for instance gisu429.gisa-halle.de via iallocator hail: gisu818.gisa-halle.de
2014-04-15 09:30:31,423: ganeti-masterd pid=22310/Jq18/Job890543/I_CREATE INFO Selected nodes for instance gisu430.gisa-halle.de via iallocator hail: gisu817.gisa-halle.de
2014-04-15 09:32:16,040: ganeti-masterd pid=22310/Jq25/Job890544/I_CREATE INFO Selected nodes for instance gisu431.gisa-halle.de via iallocator hail: gisu818.gisa-halle.de

Right after creation, running hbal will calculate, that some instances should migrate:

2014-04-15 09:39:02,418: ganeti-masterd pid=22310/ClientReq13 INFO New job with id 890555, summary: INSTANCE_MIGRATE(gisu429.gisa-halle.de)

<b>Please provide any additional information below.</b>

It seems to happen often in a "unbalanced" cluster, i.e. after adding a new node. But I'm shure the capacity of the "old" nodes hasn't been reached (like hbal afterwards proves).

Thanks, Sascha.
```
Originally added on 2014-04-15 10:49:33 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Working as intended.

Ganeti only considers those tags as exclusion tags where the tag
name starts with "htools:iextags:". This is on purpose, as tags
can be used for other purposes than exclusion as well.
```
Originally added on 2014-04-15 11:49:53 +0000 UTC.

Changed State: WontFix

**Comment by Ganeti-Issues-Migrator** [comment]
```
Just to clarify, the correct way to use exclusion tags in your example would be

# gnt-instance add -t sharedfile ... --tags htools:iextags:sapsid:xxx vm1
# gnt-instance add -t sharedfile ... --tags htools:iextags:sapsid:xxx vm2

```
Originally added on 2014-04-15 11:52:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
It might be a documentation bug. We will look at the hbal and hail man pages.
```
Originally added on 2014-04-15 11:59:42 +0000 UTC.

Added Labels: Type-Documentation Priority-Medium
Changed State: New
Added to Milestone: Release2.9

**Comment by Ganeti-Issues-Migrator** [comment]
```
Reproduced, and behavior indeed does not fit documentation.
```
Originally added on 2014-04-15 12:16:23 +0000 UTC.

Added Labels: Priority-High
Removed Labels: Type-Documentation Priority-Medium
Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
It seems to be a problem with the IAllocator interface; hbal indeed behaves
as in the man page. So probably the documentation is correct, while the IAllocator interface isn't. Will investigate further.
```
Originally added on 2014-04-15 12:30:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
OK, found the cause: bug in the parsing of IAllocator requests. Sorry for trying to close the bug too early.
```
Originally added on 2014-04-15 15:27:51 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 93c6a885668575ab778c05cd875305a0666d7bc8
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Apr 15 16:49:51 2014 +0200

    Set exclusion tags correctly in requested instance
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Jose Lopes <jabolopes@google.com>

```
Originally added on 2014-04-15 16:16:16 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I believe the issue is fixed now. Can you please verify that on version 93c6a885 (of the stable-2.9 branch) everything works as expected?
```
Originally added on 2014-04-15 16:17:17 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

impressive fast bugfix! I've tested it and it works now.

Thanks, Sascha.
```
Originally added on 2014-04-16 09:10:48 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
released as part of 2.11.6
```
Originally added on 2014-09-22 15:58:21 +0000 UTC.

Changed State: Released
